src/sentences/utils.rs

```diff
@@ -49,6 +49,43 @@ pub(crate) fn parse_hms(i: &str) -> IResult<&str, NaiveTime> {
     )(i)
 }
 
+/// The number of milliseconds in a second.
+const MILLISECS_PER_SECOND: i64 = 1000;
+/// The number of milliseconds in a minute.
+const MILLISECS_PER_MINUTE: i64 = 60000;
+/// The number of milliseconds in a hour.
+const MILLISECS_PER_HOUR: i64 = 3600000;
+
+pub(crate) fn parse_duration_hms(i: &str) -> IResult<&str, Duration> {
+    map_res(
+        tuple((
+            map_res(take(2usize), parse_num::<i64>),
+            map_res(take(2usize), parse_num::<i64>),
+            map_parser(take_until(","), double),
+        )),
+        |(hour, minutes, sec)| -> core::result::Result<Duration, &'static str> {
+            if sec.is_sign_negative() {
+                return Err("Invalid time: second is negative");
+            }
+            if hour >= 24 {
+                return Err("Invalid time: hour >= 24");
+            }
+            if minutes >= 60 {
+                return Err("Invalid time: min >= 60");
+            }
+            if sec >= 60. {
+                return Err("Invalid time: sec >= 60");
+            }
```

Comment:
Also check for `< 0.0` and `NaN` (https://doc.rust-lang.org/nightly/std/primitive.f64.html#method.is_nan)